xarray/core/common.py

```
@@ -211,14 +211,17 @@ def _ipython_key_completions_(self):
         return list(set(item_lists))
 
 
-def get_squeeze_dims(xarray_obj, dim):
+def get_squeeze_dims(xarray_obj, dim, axis=None):
     """Get a list of dimensions to squeeze out.
     """
-    if dim is None:
+    if dim is None and axis is None:
```

Maybe we can raise an Error if dim and axis contradict each other?